Source/Core/Common/JsonUtil.h

```diff
@@ -24,3 +27,18 @@ picojson::array ToJsonArray(const Range& data)
 
   return result;
 }
+
+template <typename Type>
+Type ReadNumericOrDefault(const picojson::object& obj, const std::string& key,
+                          Type default_value = Type{})
+{
+  const auto it = obj.find(key);
+  if (it == obj.end())
+    return default_value;
+  if (!it->second.is<double>())
+    return default_value;
+  return static_cast<Type>(it->second.get<double>());
```

Comment:
If you're making this generic then this should maybe use `Common::SaturatingCast` for integers...?

Reply:
Was not aware of that function!  Thanks.